main.go

```diff
@@ -0,0 +1,127 @@
+package authenticated_block_store
+
+import (
+	"fmt"
+	"log"
+	"io/ioutil"
+        "strings"
+        "os"
+        "crypto/sha256"
+        "math/rand"
+        "time"
+
+)
+
+//size of each block, intended to be 4KB TODO
+const BLOCK_SIZE = 16
+
+var (
+        logFileNames []string
+        logFileHandles []*os.File
+)
+
+type logFileNode struct {
+        name string
+        hash string
+        nextHash string
+        next *logFileNode
+        previous *logFileNode //might not be needed
+}
+
+type dataBase struct {
+        head *logFileNode
+        tail *logFileNode
+        merkleRoot string
+        // other components
+}
+
+func initializeDB() *dataBase {
+        return &dataBase{
+                head:nil,
+                tail:nil,
+                merkleRoot:"",
+        }
+}
+
+func readLogFile(input *os.File) *logFileNode{
+        //do sanity checks
+        var newNode logFileNode
+        var err error
+        newNode.name = input.Name()
+        b := make ([]byte, 4)
+        _, err = input.ReadAt(b, 0)
+        check (err)
+        newNode.hash = string(b)
+        _, err = input.ReadAt(b, 0)
+        check (err)
+        return &newNode
+}
+
+func check(e error) {
```

Comment:
If you can, use the standard library here: https://godoc.org/gopkg.in/check.v1

Reply:
Probably gonna remove it all, and use Testing only. But I'll look into check too. Thanks!